metricbeat/module/kafka/consumergroup/query.go

```diff
@@ -38,51 +39,55 @@ func fetchGroupInfo(
 
 	debugf("known consumer groups: ", groups)
 
+	assignments, err := fetchGroupAssignments(b, groups)
+	if err != nil {
+		logp.Err("failed to fetch kafka group assignments: %v", err)
+		return err
+	}
+	if len(assignments) == 0 {
+		return nil
+	}
+
 	wg := sync.WaitGroup{}
-	results := make(chan result, len(groups))
-	for _, group := range groups {
-		group := group
+	results := make(chan result, len(assignments))
+
+	waiting := 0
+	for group, topics := range assignments {
+		_partitions := make(map[string][]int32)
+		for topic, partitions := range topics {
+			if topicsFilter == nil || topicsFilter(topic) {
+				for partitionID := range partitions {
+					if _, ok := _partitions[topic]; !ok {
+						_partitions[topic] = make([]int32, 0)
+					}
```

Comment:
This if-clause is not required. `_partitions[topic]` returns `nil`, if topic is not known. `append(([]int32)(nil), value)` returns a new `[]int32{value}`.

Reply:
fixed: optimalized the code generate partition array

Reply:
fixed: return the error if has